hazelcast-tpc-engine/src/test/java/com/hazelcast/internal/tpc/util/OSTest.java

```diff
@@ -41,11 +41,21 @@ public void test_isLinux0() {
 
     @Test
     public void test_linuxMajorVersion0() {
-        assertEquals(5, OS.linuxMajorVersion0("5.16.12-200.fc35.x86_64"));
+
+        if (OS.isLinux()) {
```

Comment:
But the test fails on Mac OS as it is.

Reply:
I forgot about the first part of the function:

```
  static int linuxMajorVersion0(String version) {
        if (!isLinux()) {
            return -1;
        }

        String[] versionTokens = version.split("\\.");
        try {
            return Integer.parseInt(versionTokens[0]);
        } catch (NumberFormatException e) {
            return -1;
        }
    }
```

The idea was to make it a pure function, which apparently didn't happen. This would make it a pure function

```
 static int linuxMajorVersion0(String version, boolean isLinux) {
        if (!isLinux) {
            return -1;
        }

        String[] versionTokens = version.split("\\.");
        try {
            return Integer.parseInt(versionTokens[0]);
        } catch (NumberFormatException e) {
            return -1;
        }
    }
```

Reply:
I am fine with any approach.

Reply:
With the pure approach, you can properly test the version extraction on any platform. You can still extract "5" from "5.16.12-200.fc35.x86_64".